### DISCUSSION TITLE
关于使用DDNS后的域名解析问题

### DISCUSSION THREAD

**Discussion by Huiaei** [messense/aliyundrive-webdav#352] [Q&A] [answered]
我经过测试观察到，程序好像只允许指定的ip进行访问，比如我设置主机为`192.168.1.2`，那它只允许通过这个ip访问，我在部署的设备上都无法通过`127.0.0.1`的ip进行访问，提示“已拒绝连接”。

我设置主机名为DDNS解析的域名，发现程序启动时会将域名转为IP地址，一开始肯定都是没有问题，但是当IP变动，DDNS刷新了域名记录，IP记录变了，所以当初程序解析后绑定的IP地址与现在对不上，导致会拒绝访问。

**Comment by messense** [comment, marked as answer]
设置为 0.0.0.0

**Reply by Huiaei** [reply]
设置为 0.0.0.0 后，发现 ipv6 不能访问。
设置为 [::] 后，发现 ipv4 又不能访问了。

**Reply by messense** [reply]
用 `::`

**Reply by Huiaei** [reply]
是使用的 `::` ，设置后 ipv4 访问就不行了。

![启动脚本](https://onemanager.huiaei.vercel.app/%F0%9F%96%BCImages/ImgURL/Snipaste_2022-04-03_15-53-56.png
)

![运行中](https://onemanager.huiaei.vercel.app/%F0%9F%96%BCImages/ImgURL/3cd2ee44a5cff064e18c9700cf92e45f.png
)

![IPV4](https://onemanager.huiaei.vercel.app/%F0%9F%96%BCImages/ImgURL/b6334d4931508ede580b56aa86746929.png
)

![IPV6](https://onemanager.huiaei.vercel.app/%F0%9F%96%BCImages/ImgURL/382f4bffd5825ed0dc3e02ba2bc7cb23.png
)

**Reply by messense** [reply]
Windows dual stack 没试过，不用 Windows 所以不会进一步排查解决，有能力可以自行排查解决。
